lib/tower.ex

```diff
@@ -573,7 +573,30 @@ defmodule Tower do
     Application.fetch_env!(:tower, :reporters)
   end
 
-  defp async(fun) do
+  def async(fun) do
+    in_unlinked_task(fn ->
+      try do
+        fun.()
+      rescue
+        exception ->
+          in_unlinked_task(fn ->
+            raise ReportEventError,
```

Comment:
This now I realize doesn't make sense as is: raising an error called ReporEventError from a function that is called `async`. It can be called from a reporter report_event, but not necessarily from this point of view.